litestar/plugins/flash.py

```diff
@@ -0,0 +1,49 @@
+from dataclasses import dataclass
+from typing import Any, Mapping
+
+from litestar.config.app import AppConfig
+from litestar.connection import ASGIConnection
+from litestar.contrib.minijinja import MiniJinjaTemplateEngine
+from litestar.plugins import InitPluginProtocol
+from litestar.template import TemplateConfig
+from litestar.template.base import _get_request_from_context
+from litestar.utils.scope.state import ScopeState
+
+
+@dataclass
+class FlashConfig:
+    """Configuration for Flash messages."""
+
+    template_config: TemplateConfig
+
+
+class FlashPlugin(InitPluginProtocol):
+    """Flash messages Plugin."""
+
+    def __init__(self, config: FlashConfig):
+        """Initialize the plugin."""
+        self.config = config
+
+    def on_app_init(self, app_config: AppConfig) -> AppConfig:
+        """Register the message callable on the template engine instance."""
+        if isinstance(self.config.template_config.engine_instance, MiniJinjaTemplateEngine):
+            from litestar.contrib.minijinja import _transform_state
```

Comment:
is it okay to be calling a private method inside a public one?

Reply:
honestly I'm not real happy about it, but the minijinja implementation has that particularity of needing that "tweak", I'm really happy if there is another way, @provinzkraut had something similar a few commits above, I find this solution a little bit simpler but have no real preference of one over another, maybe someone would come up with something better

Reply:
Well this interface isn't great, I just did this as a poc to fix the bug :grimacing:

Reply:
in essence this is kind of minijina's implementation fault (another stuff of mine :-1: ) I have to resort to this, maybe a more elegant solution would be to rethink how minijinja is implemented in order not to have to resort to swith and private import, but I failed at finding a way to be fair